data/fintype.lean

```diff
@@ -586,4 +586,69 @@ lemma fintype.card_equiv [fintype α] [fintype β] (e : α ≃ β) :
   fintype.card (α ≃ β) = (fintype.card α).fact :=
 fintype.card_congr (equiv_congr (equiv.refl α) e) ▸ fintype.card_perm
 
-end equiv
\ No newline at end of file
+end equiv
+
+section choose_unique
+open fintype
+open equiv
+
+variables [fintype α] [decidable_eq α] (p : α → Prop) [decidable_pred p]
+
+def choose_unique_x (p_eu : ∃! a : α, p a) : {a // p a} :=
+begin
+  apply @trunc.lift (α ≃ fin (card α)) _ _ _ (equiv_fin α),
+  intro equiv_fin,
+  let G : fin (card α) → Prop := λ f, p (equiv_fin.inv_fun f),
+  have G_exists : {f : fin (card α) // G f},
+  { apply @encodable.choose_x (fin (card α)) G,
+  rcases p_eu with ⟨p_w, ⟨p_h, p_u⟩⟩,
+  existsi equiv_fin.to_fun p_w,
+  change p (equiv_fin.inv_fun (equiv_fin.to_fun p_w)),
+  rw equiv_fin.left_inv, assumption },
+  { existsi (equiv_fin.inv_fun G_exists.val),
+  exact G_exists.property },
+  have all_equal : (Π a b : {a // p a}, a = b),
+  { rintros ⟨a_w, a_h⟩ ⟨b_w, b_h⟩,
+  rcases p_eu with ⟨p_w, ⟨p_h, p_u⟩⟩,
+  congr, rw [p_u a_w a_h, p_u b_w b_h] },
+  intros,
+  apply all_equal
+end
+
+def choose_unique (p_eu : ∃! a : α, p a) : α := (choose_unique_x p p_eu).1
+
+def choose_unique_spec {p_eu : ∃! a : α, p a} : p (choose_unique p p_eu) :=
+(choose_unique_x p p_eu).2
+
+end choose_unique
+
+section bijection_inverse
+open function
+
+variables [fintype α] [decidable_eq α]
+variables [fintype β] [decidable_eq β]
+variables {f : α → β} 
+
+def bij_inv (f_bij : bijective f) (b : β) : α :=
+begin
+  apply choose_unique (λ a, f a = b),
+  cases f_bij.right b,
+  existsi w,
+  constructor,
+  { exact h },
+  { intro,
+  rw ← h,
```

Comment:
take care of your indendation. Also give the variables introduced with `intro` a name if you use them later.